### DISCUSSION TITLE
Registering with just email and password (without username)

### DISCUSSION THREAD

**Discussion by andriyndev** [serverpod/serverpod#823] [Q&A] [answered]
Is it possible to register a user with just email and password without username? I'm working on an app which registration form doesn't suppose to have the username field but the `createAccountRequest()` method of `EmailAuthController` requires it.

**Comment by yahu1031** [comment, marked as answer]
In that case you can hardcore the username by manipulating the mail id. 

EG: hi2minnu@gmail.com -> hi2minnu 

Something like this as you wish. As the user mails will be unique the username will be also unique. Or else you can do this. 

Username: <APP_PREFIX>ID

EG: App name - My App

Username : MA123

**Reply by andriyndev** [reply]
Thank you! However, why the field is mandatory anyway?

**Reply by yahu1031** [reply]
Maybe usernames are considered unique IDs. And BD transactions will be easy with that.

**Reply by vlidholt** [reply]
User names must not be unique. In principle, you can just add an empty string for the user name.

**Reply by andriyndev** [reply]
Thank you!

**Reply by Frank1016** [reply]
Hi @vlidholt, I just double checked the creatAccountRequest() method, if I use an empty string or a blank space for the user name, there will be no response from the method at all. However, if I provide any normal string for the user name field, it will work fine. I'm on the latest ServerPod 1.1.1

**Reply by vlidholt** [reply]
@Frank1016, thanks for clarifying. We will look over this for version 2!
